lib/selinux/selinux_linux.go

```diff
@@ -77,33 +78,38 @@ func CheckConfiguration() error {
 		return trace.Errorf("SELinux mode is not enforcing, SELinux will not constrain anything")
 	}
 
-	selinuxType := readConfig(selinuxTypeTag)
-	if selinuxType == "" {
-		return trace.NotFound("could not find SELinux type")
+	selinuxType, err := readConfig(selinuxTypeTag)
+	if err != nil {
+		return trace.Wrap(err, "failed to find SELinux type")
 	}
 	selinuxDir := filepath.Join("/var/lib/selinux", selinuxType, "active/modules")
 
 	var moduleInstalled, moduleDisabled, modulePermissive bool
-	err := filepath.WalkDir(selinuxDir, func(path string, d fs.DirEntry, err error) error {
+	err = filepath.WalkDir(selinuxDir, func(path string, d fs.DirEntry, err error) error {
 		if err != nil {
-			return err
+			return trace.Wrap(err, "failed to access %q", path)
 		}
 
 		name := d.Name()
-		if strings.Contains(name, moduleName) {
+		if !strings.Contains(name, moduleName) {
+			return nil
+		}
+
+		if d.IsDir() {
 			moduleInstalled = true
 			if name == permissiveModuleName {
 				modulePermissive = true
 				// if the module is permissive, we also know it's installed
 				// so we can end the walk
 				return filepath.SkipAll
 			}
-			if filepath.Base(filepath.Dir(path)) == "disabled" {
-				moduleDisabled = true
-				// if the module is disabled, we also know it's installed
-				// so we can end the walk
-				return filepath.SkipAll
-			}
+			// if the module is disabled, an empty file with the module's
+			// name will exist in the "disabled" dir
+		} else if filepath.Base(filepath.Dir(path)) == "disabled" {
```

Comment:
nit: reduce nesting a bit by removing the `else` and returning from the `if d.IsDir()` in all cases

Reply:
I'm not really sure how I'd reduce nesting tbh, as I still need the conditional on line 108; returning at the end of the `if d.IsDir()` conditional wouldn't change the amount of nesting at all